### PR TITLE
Rename node status to node health in NodeLifecycleController

### PR DESCRIPTION
**Pull request by wangzhen127**
**What this PR does / why we need it**:
Since we are going to treat both node status and node lease as node heartbeat/health signals, this PR makes the renmae changes, so that the follow-up PR (#69241) is easier to review. This PR also updates the saved node status and node health map to use pointers to avoid extra copy.

This is part of [KEP-0009](https://github.com/kubernetes/community/blob/master/keps/sig-node/0009-node-heartbeat.md) and feature [#589](https://github.com/kubernetes/features/issues/589).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by wangzhen127** [comment]
Also adding the approver:
/assign @gmarek

File: pkg/controller/nodelifecycle/node_lifecycle_controller.go [hunk 1]
```diff
@@ -128,14 +128,14 @@ const (
...
+	// The amount of time the nodecontroller should sleep between retrying node health updates
 	retrySleepTime = 20 * time.Millisecond
 )
 
-type nodeStatusData struct {
+type nodeHealthData struct {
 	probeTimestamp           metav1.Time
 	readyTransitionTimestamp metav1.Time
-	status                   v1.NodeStatus
+	status                   *v1.NodeStatus
```
Comment (yujuhong) [inline pkg/controller/nodelifecycle/node_lifecycle_controller.go]: The commit message is slightly misleading since you also changed this to a pointer.

**Review by yujuhong — COMMENTED** [review]
/lgtm

File: pkg/controller/nodelifecycle/node_lifecycle_controller.go (hunk 1 above)
Comment (wangzhen127) [reply pkg/controller/nodelifecycle/node_lifecycle_controller.go]: Updated the commit message to include the pointer change now. :)

**Comment by wangzhen127** [comment]
nit change after @yujuhong's lgtm: savednodeHealth -> saved**N**odeHealth

**Comment by wojtek-t** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**